pytket/pytket/wasm/wasm.py

```diff
@@ -16,24 +16,182 @@
+                                entry.field_str.tobytes().decode()
+                            )
+
+                # read in map of function signatures to function names
+                elif cur_sec_data.id == SEC_FUNCTION:
+                    self._function_types = cur_sec_data.payload.types
+
+            for i, x in enumerate(self._function_names):
+
+                # check for only i32 type in parameters and return values
+                supported_function = True
+                for t in self._function_signatures[self._function_types[i]][
+                    "parameter_types"
+                ]:
+                    if t != "i32":
+                        supported_function = False
+                for t in self._function_signatures[self._function_types[i]][
+                    "return_types"
+                ]:
+                    if t != "i32":
+                        supported_function = False
+
+                if supported_function:
+                    self._functions[x] = (
+                        len(
+                            self._function_signatures[self._function_types[i]][
+                                "parameter_types"
+                            ]
+                        ),
+                        len(
+                            self._function_signatures[self._function_types[i]][
+                                "return_types"
+                            ]
+                        ),
+                    )
+
+                if not supported_function:
+                    self._unsupported_function.append(x)
+
     def __str__(self) -> str:
         """str representation of the wasm file"""
         return self._wasmuid
+
+    def __repr__(self) -> str:
+        """str representation of the containment of the wasm file"""
+        result = f"Functions in wasm file with the uid {self._wasmuid}:\n"
+        for x in self._functions:
+            result += f"function '{x}' with {self._functions[x][0]} i32 parameter(s)"
+            result += f" and {self._functions[x][1]} i32 return value(s)\n"
+
+        for x in self._unsupported_function:
+            result += (
+                f"unsupported function with unvalid parameter or result type: '{x}' \n"
+            )
+
+        return result
+
+    def check_function(
+        self, function_name: str, number_of_parameters: int, number_of_returns: int
+    ) -> bool:
+        """checks a given function name and signature if it is included
+        :param function_name: name of the function that is checked
+        :type function_name: str
+        :param number_of_parameters: number of i32 parameters of the function
+        :type number_of_parameters: int
+        :param number_of_returns: number of i32 return values of the function
+        :type number_of_returns: int
+        :return: true if the signature and the name of the function is correct"""
+        if not self._check_file:
+            return True
```

Comment:
Perhaps this should raise an exception rather than return True?

Reply:
The user can't choose if the check is executed. I was thinking about if this hole option to opt out of the check should be removed...

Reply:
I think we don't need it, it is only making thinks more complicated and in-transparent for the user.